src/write.c

```diff
@@ -160,11 +161,11 @@ static avifEncoderItem * avifEncoderDataCreateItem(avifEncoderData * data, const
     if (!avifArrayCreate(&item->mdatFixups, sizeof(avifOffsetFixup), 4)) {
         goto error;
     }
-    ++data->lastItemID;
     return item;
 
 error:
     avifCodecEncodeOutputDestroy(item->encodeOutput);
+    --data->lastItemID;
```

Comment:
This is fine. Another option to avoid the decrement in the error path would be `item->id = data->lastItemID + 1;` at line 155.